### DISCUSSION TITLE
Validation error does not go away after field ceases to be required

### DISCUSSION THREAD

**Discussion by jozef-mikusinec-kiwicom** [react-hook-form/react-hook-form#10560] [Q&A] [answered]
### Version Number

7.45.0

### Codesandbox/Expo snack

https://codesandbox.io/s/naughty-star-7d8jyq?file=/src/index.js

### Steps to reproduce

0. In the sandbox, type something into the text field, and then delete it. The error "Required field" will appear.
1. Tick the checkbox. The error does not go away.
2. You can repeat step zero and the error will go away.

### Expected behaviour

The error should disappear in step 1, but only does so in step 2.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by leapful** [comment, marked as answer]
@jozef-mikusinec-kiwicom 

`Controller` component will handle validation at field level, therefore it only triggers validation of the target field, in this case it's your `checkbox` element.

You need to trigger manually validation on your `textField` element whenever `checkbox` is changed to achieve the expected result.

Working demo: https://codesandbox.io/s/spring-thunder-tthwhd?file=/src/index.js

**Reply by jozef-mikusinec-kiwicom** [reply]
I guess this is an answer consistent with what the maintainers of react-hook-form would consider correct, so I'm marking this as answer.

Despite that, I know that yup checks the validity of the entire form each time any field is changed, and I believe it should still be react-hook-form that should handle the revalidation of all other fields as well, on its own, automatically.

Thank you for your answer.
